tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionDebugging.cs

```diff
@@ -49,24 +51,20 @@ public static void Initialize()
             {
                 Log.Warning("Third party modules load has failed. Disabling Exception Debugging.");
                 _isDisabled = true;
+                return false;
             }
-            else
-            {
-                InitSnapshotsSink();
-                ExceptionTrackManager.Initialize();
-                LifetimeManager.Instance.AddShutdownTask(Dispose);
-            }
+
+            InitSnapshotsSink();
+            _exceptionTrackManager = ExceptionTrackManager.Create(Settings);
+            LifetimeManager.Instance.AddShutdownTask(Shutdown);
```

Comment:
I think we should get rid of it to avoid leakage. Now that we enable/disable ER/DI/CO dynamically, we should avoid subscribing to global events or at least unsubscribe upon destruction
(This comment is relevant to all `LifetimeManager.Instance.AddShutdownTask` calls)

We should validate that `DebuggerManager` does indeed take care of it on itself upon shutdown

Reply:
Fixed in 801841e1b27412fa577c5a1f60a7288a7d01906b